tests/annotator/annotator.cpp

```diff
@@ -0,0 +1,2131 @@
+        model->component("component2")->variable("variable2"));
+
+    EXPECT_EQ("b4da62", annotator->assignId(connection2, libcellml::CellMLElement::CONNECTION));
+    EXPECT_EQ("b4da62", libcellml::Variable::equivalenceConnectionId(connection2.first, connection2.second));
+
+    EXPECT_EQ("b4da63", annotator->assignId(mapping2, libcellml::CellMLElement::MAP_VARIABLES));
+    EXPECT_EQ("b4da63", libcellml::Variable::equivalenceMappingId(mapping2.first, mapping2.second));
+}
+
+TEST(Annotator, assignAllIdsBlankModel)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(modelStringNoIds);
+    auto annotator = libcellml::Annotator::create();
+
+    annotator->assignAllIds(model);
+    EXPECT_EQ(size_t(0), annotator->errorCount());
+
+    libcellml::ModelPtr nullModel;
+    annotator->assignAllIds(nullModel);
+    EXPECT_EQ(size_t(0), annotator->errorCount());
+}
+
+TEST(Annotator, assignAllIdsNonblankModel)
+{
+    auto parser = libcellml::Parser::create();
+    auto printer = libcellml::Printer::create();
+    auto model = parser->parseModel(modelStringUniqueIds);
+    auto annotator = libcellml::Annotator::create();
+
+    annotator->assignAllIds(model);
+    EXPECT_EQ(size_t(0), annotator->errorCount());
+    EXPECT_EQ(modelStringUniqueIds, printer->printModel(model));
+}
+
+TEST(Annotator, clearAllIdsNoModelPresent)
+{
+    std::vector<std::string> expectedErrors = {
+        "This Annotator object does not have a model to work with."};
+
+    auto parser = libcellml::Parser::create();
+    auto annotator = libcellml::Annotator::create();
+
+    annotator->clearAllIds();
+    EXPECT_EQ(size_t(1), annotator->errorCount());
+    EXPECT_EQ(expectedErrors[0], annotator->error(0)->description());
+}
+
+TEST(Annotator, clearAllIdsNullModelPresent)
+{
+    std::vector<std::string> expectedErrors = {
+        "This Annotator object does not have a model to work with."};
+
+    auto parser = libcellml::Parser::create();
+    auto annotator = libcellml::Annotator::create();
+
+    libcellml::ModelPtr nullModel;
+    annotator->clearAllIds(nullModel);
+    EXPECT_EQ(size_t(1), annotator->errorCount());
+    EXPECT_EQ(expectedErrors[0], annotator->error(0)->description());
+}
+
+TEST(Annotator, clearAllIdsModelPresent)
+{
+    std::vector<std::string> expectedErrors = {
+        "This Annotator object does not have a model to work with."};
+
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(modelStringDuplicateIds);
+    auto annotator = libcellml::Annotator::create();
+
+    annotator->clearAllIds(model);
+    EXPECT_EQ(size_t(0), annotator->errorCount());
+
+    libcellml::AnyItem item = annotator->item("duplicateId");
+    EXPECT_EQ(libcellml::CellMLElement::UNDEFINED, item.first);
+}
+
+TEST(Annotator, assignIdInvalidType)
+{
+    const std::string e = "The given type 'reset' is invalid for this method.";
```

Comment:
This error message doesn't help me. I would expect to know why it is invalid.

Reply:
b386607e